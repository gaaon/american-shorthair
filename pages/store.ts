import { observable } from 'mobx';
import { NextPageContext } from 'next-server/dist/lib/utils';
import cookie from 'cookie';
import Cookies from 'js-cookie';

const createSettingStore = (ctx: NextPageContext) => {
  let isHomeAnimActive: string | undefined;

  if (ctx.req) {
    const headers = ctx.req.headers;
    const ctxCookie = cookie.parse(headers.cookie || '');
    isHomeAnimActive = ctxCookie['isHomeAnimActive'];
  } else {
    isHomeAnimActive = Cookies.get('isHomeAnimActive');
  }

  return observable({
    isHomeAnimActive,
  });
};

export const createRootStore = (ctx: NextPageContext) => observable({
  setting: createSettingStore(ctx),
});

