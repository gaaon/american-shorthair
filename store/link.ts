import {LinkItem} from '../types';
import {observable} from 'mobx';

import { faFlask } from '@fortawesome/free-solid-svg-icons';
import { faSmileWink, faLightbulb } from '@fortawesome/free-regular-svg-icons';

export interface LinkStore {
  internalLinks: Array<LinkItem>
}

export const createLinkStore = (): LinkStore => observable({
  internalLinks: [
    {link: '/intro', label: '소개', icon: faSmileWink},
    {link: '/service', label: '서비스', icon: faLightbulb},
    {link: '/laboratory', label: '실험실', icon: faFlask},
    // {link: '/blog', label: '블로그', icon: 'fas fa-blog'},
  ],
});
