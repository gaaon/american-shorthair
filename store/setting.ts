import { action, observable } from 'mobx';

import { NextPageContext } from 'next';

import cookie from 'cookie';
import Cookies from 'js-cookie';

export interface Setting {
  isHomeAnimActive: boolean;
}

export interface SettingStore extends Setting {
  toggleAnimActive: () => void;
}

export const hydrateSetting = (ctx: NextPageContext): Setting => {
  let isHomeAnimActive: string | undefined;

  if (ctx.req) {
    const headers = ctx.req.headers;
    const ctxCookie = cookie.parse(headers.cookie || '');
    isHomeAnimActive = ctxCookie['isHomeAnimActive'];
  } else {
    isHomeAnimActive = Cookies.get('isHomeAnimActive');
  }

  return {
    isHomeAnimActive: isHomeAnimActive === undefined || isHomeAnimActive === 'true',
  };
};

export const createSettingStore = ({ isHomeAnimActive }: Setting): SettingStore => {
  const store = observable({
    isHomeAnimActive: isHomeAnimActive,
    toggleAnimActive() {
      store.isHomeAnimActive = !store.isHomeAnimActive;
      Cookies.set('isHomeAnimActive', (store.isHomeAnimActive) + '');
    },
  }, {
    toggleAnimActive: action,
  });

  return store;
};
