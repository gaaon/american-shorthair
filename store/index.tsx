import React, { createContext } from 'react';

import { useLocalStore } from 'mobx-react';

import {AppContext} from 'next/app';

import {createSettingStore, hydrateSetting, SettingStore} from './setting';
import {createLinkStore, LinkStore} from './link';
// import {createLinkStore} from './link';

export const StoreContext = createContext<RootStore | null>(null);

export interface RootStore {
  setting: SettingStore;
  link: LinkStore;
}

export const createData = (appCtx: AppContext): RootStore => {
  return {
    setting: createSettingStore(hydrateSetting(appCtx.ctx)),
    link: createLinkStore(),
  };
};


let rootStore: RootStore;
export const initializeData = (initialData = rootStore || {}): RootStore => {
  let { setting: { isHomeAnimActive = true } = {}, link } = initialData;

  return {
    setting: createSettingStore({isHomeAnimActive}),
    link,
  }
};

export const InjectStoreContext: React.FC<{initialData: RootStore}> = ({ children, initialData }) => {
  rootStore = useLocalStore(() => initializeData(initialData));

  if (rootStore == null) return null;

  return (
    <StoreContext.Provider value={rootStore}>{children}</StoreContext.Provider>
  )
};

export const useStore = () => {
  const store = React.useContext(StoreContext);
  if (!store) {
    throw new Error("useStore: !store, did you forget StoreProvider?");
  }
  return store;
};