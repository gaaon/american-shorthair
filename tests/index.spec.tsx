import React from 'react';

import renderer from 'react-test-renderer';

import Home from '../pages';

import { initializeData, InjectStoreContext } from '../store';
import { createLinkStore } from '../store/link';

it('renders correctly', () => {
  const initialStoreData = initializeData();
  initialStoreData.link = createLinkStore();

  const tree = renderer
    .create(
      <InjectStoreContext initialData={initialStoreData}>
        <Home/>
      </InjectStoreContext>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});