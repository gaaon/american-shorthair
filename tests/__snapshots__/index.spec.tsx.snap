// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="d-flex flex-column min-h-100 h-100"
  style={
    Object {
      "overflowX": "hidden",
    }
  }
>
  <div
    className="main container"
  >
    <div
      className="h-100 flex-column flex-md-row row"
    >
      <div
        className="left  col"
      >
        <div
          className="h-100 flex-column justify-content-end justify-content-sm-center row"
        >
          <div
            className="text-right flex-grow-0 col"
          >
            <div
              className="justify-content-end row"
            >
              <div
                className="col"
              >
                <img
                  alt="my photo"
                  className="avatar"
                  src="/static/img/my_photo.jpeg"
                />
              </div>
            </div>
            <h1
              className="display-4 d-none d-lg-block text-weight-bold"
            >
               
              <strong>
                 
                <small
                  className="text-orange"
                >
                  모든 코드에 대한
                </small>
                 
                <br />
                taawoo의 페이지
                 
              </strong>
               
            </h1>
            <p
              className="h2 d-block d-lg-none"
            >
               
              <small
                className="text-orange"
              >
                모든 코드에 대한
              </small>
               
              <br />
              taawoo의 페이지
               
            </p>
          </div>
        </div>
      </div>
      <div
        className="right  col"
      >
        <div
          className="h-100 flex-column justify-content-center row"
        >
          <div
            className="d-none d-sm-block flex-grow-0 col"
            style={
              Object {
                "minHeight": 24,
              }
            }
          />
          <div
            className="col"
          >
            <div
              className="h-100 flex-column justify-content-sm-center justify-content-top row"
            >
              <div
                className="flex-grow-0 col"
              >
                <ul
                  className="menuList"
                >
                  <li
                    className="mt-4"
                  >
                    <p
                      className="h4"
                    >
                      <a
                        className="menuItem"
                        href="/intro"
                        onClick={[Function]}
                      >
                        <span
                          className="d-inline-block text-center"
                          style={
                            Object {
                              "minWidth": 50,
                            }
                          }
                        />
                        <span
                          style={
                            Object {
                              "letterSpacing": ".1em",
                            }
                          }
                        >
                          소개
                        </span>
                      </a>
                    </p>
                  </li>
                  <li
                    className="mt-4"
                  >
                    <p
                      className="h4"
                    >
                      <a
                        className="menuItem"
                        href="/service"
                        onClick={[Function]}
                      >
                        <span
                          className="d-inline-block text-center"
                          style={
                            Object {
                              "minWidth": 50,
                            }
                          }
                        />
                        <span
                          style={
                            Object {
                              "letterSpacing": ".1em",
                            }
                          }
                        >
                          서비스
                        </span>
                      </a>
                    </p>
                  </li>
                  <li
                    className="mt-4"
                  >
                    <p
                      className="h4"
                    >
                      <a
                        className="menuItem"
                        href="/laboratory"
                        onClick={[Function]}
                      >
                        <span
                          className="d-inline-block text-center"
                          style={
                            Object {
                              "minWidth": 50,
                            }
                          }
                        />
                        <span
                          style={
                            Object {
                              "letterSpacing": ".1em",
                            }
                          }
                        >
                          실험실
                        </span>
                      </a>
                    </p>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className="flex-grow-0 col"
          >
            <div
              className="form-check form-check-inline"
            >
              <label
                className="form-check-label"
                htmlFor="anim-active"
              >
                 애니메이션 끄기 
              </label>
              <input
                checked={false}
                className="form-check-input ml-2"
                id="anim-active"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="backgroundMain"
  >
    <div
      className="h-100 container"
    >
      <div
        className="h-100 flex-column justify-content-center row"
      >
        <div
          className="flex-grow-0 col"
        >
          <h2
            className="display-4 text-center"
          >
            좋아요
            <br />
            갑시다!
          </h2>
          <div
            className="text-center"
          >
             
            <del>
               다양한 멘트 준비중..! 
            </del>
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
